Marlin/Marlin_main.cpp

```diff
@@ -2702,30 +2708,80 @@ inline void gcode_G28() {
-                // Include + for column alignment
-                else
-                  SERIAL_PROTOCOLCHAR(' ');
-                SERIAL_PROTOCOL_F(diff, 5);
-              } // xx
-              SERIAL_EOL;
-            } // yy
+
+          SERIAL_PROTOCOLPGM(" \nCorrected Bed Topography: \n");
+          for (int yy = auto_bed_leveling_grid_points - 1; yy >= 0; yy--) {
+            for (int xx = 0; xx < auto_bed_leveling_grid_points; xx++) {
+              int ind = yy * auto_bed_leveling_grid_points + xx;
+              float diff = eqnBVector[ind] - min_diff;
+              if (diff >= 0.0)
+                SERIAL_PROTOCOLPGM(" +");   // Include + for column alignment
+              else
+                SERIAL_PROTOCOLCHAR(' ');
+              SERIAL_PROTOCOL_F(diff, 5);
+            } // xx
             SERIAL_EOL;
-          }
+          } // yy
+          SERIAL_EOL;
+
+          /*
+          SERIAL_PROTOCOLPGM(" \nBed Topography in new coordinats:\n");
+                for (int yy = auto_bed_leveling_grid_points - 1; yy >= 0; yy--) {
+                  for (int xx = 0; xx < auto_bed_leveling_grid_points; xx++) {
+                    int ind = yy * auto_bed_leveling_grid_points + xx;
+
+                    vector_3 probe_point = vector_3(eqnAMatrix[ind + 0 * abl2], eqnAMatrix[ind + 1 * abl2], eqnBVector[ind]);
+                    probe_point.apply_rotation(inverse_bed_level_matrix);
+                    float diff = probe_point.z;
+                    if (diff >= 0.0)
+                      SERIAL_PROTOCOLPGM(" +");
+                    // Include + for column alignment
+                    else
+                      SERIAL_PROTOCOLCHAR(' ');
+                    SERIAL_PROTOCOL_F(diff, 5);
+                  } // xx
+                  SERIAL_EOL;
+                } // yy
+                SERIAL_EOL;
+          */
+          SERIAL_PROTOCOLPGM(" \nCorrected Bed Topography in new coordinats:\n");
+                for (int yy = auto_bed_leveling_grid_points - 1; yy >= 0; yy--) {
+                  for (int xx = 0; xx < auto_bed_leveling_grid_points; xx++) {
+                    int ind = yy * auto_bed_leveling_grid_points + xx;
+                    vector_3 probe_point = vector_3(eqnAMatrix[ind + 0 * abl2], eqnAMatrix[ind + 1 * abl2], eqnBVector[ind]);
+                    probe_point.apply_rotation(inverse_bed_level_matrix);
+                    float diff = probe_point.z - rot_min_diff;
+                    if (diff >= 0.0)
+                      SERIAL_PROTOCOLPGM(" +");
+                    // Include + for column alignment
+                    else
+                      SERIAL_PROTOCOLCHAR(' ');
+                    SERIAL_PROTOCOL_F(diff, 5);
+                  } // xx
+                  SERIAL_EOL;
+                } // yy
+                SERIAL_EOL;
+
+          SERIAL_PROTOCOLPGM(" \nHeight from Bed to Nozzle : \n");
+          SERIAL_PROTOCOLPGM(" (+) is airprinting, (-) is touch under bed surface \n");
```

Comment:
The phrase "is touch under bed surface" is awkward. Also the description here is misleading. A height of +0.06 is not "airprinting." It's enough to say "(+) above bed, (-) below bed."